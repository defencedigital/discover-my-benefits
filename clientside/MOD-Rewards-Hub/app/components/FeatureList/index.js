import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const FL = props => (
  <div className="feature-list">
    <h3>{props.title}</h3>
    <ul className="check-list">
      {props.items.map(item => (
        <li className="check-list-item" key={`${item}`}>
          {item}
        </li>
      ))}
    </ul>
    <Link className="btn btn-lg btn-secondary" to={props.link}>
      {props.buttonText}
    </Link>
  </div>
);

FL.propTypes = {
  title: PropTypes.string,
  items: PropTypes.array,
  link: PropTypes.string,
  buttonText: PropTypes.string,
};

export default FL;
