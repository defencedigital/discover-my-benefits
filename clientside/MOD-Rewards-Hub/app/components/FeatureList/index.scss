.feature-list {
  background-color: $brand-primary;
  color: $text-color-invert;
  padding: 40px;
  border-radius: 5px;
  margin: 40px 0 60px;
}

.feature-list .check-list {
  list-style-type: none;
  padding: 20px 40px;
  margin-bottom: 20px;
}

.feature-list .check-list-item {
  position: relative;
  margin-bottom: 20px;
  font-size: 1.6rem;
  line-height: 2.6rem;

  &:last-child {
    margin-bottom: 0;
  }

  &:before {
    content: '';
    position: absolute;
    left: -40px;
    top: 0;
    width: 25px;
    height: 25px;
    background-image: url('/' + $service + '/tick.svg');
    background-size: 100%;
  }
}

