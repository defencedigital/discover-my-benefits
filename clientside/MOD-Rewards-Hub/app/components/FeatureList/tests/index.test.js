import React from 'react';
import { fromJS } from 'immutable';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import FeatureList from '../index';

Enzyme.configure({ adapter: new Adapter() });

const mockedProps = fromJS({
  title: 'title',
  items: ['one', 'two'],
  link: '/link',
});

describe('FeatureList component', () => {
  it('matches the snapshot', () => {
    const html = shallow(<FeatureList {...mockedProps.toJS()} />);
    expect(html).toMatchSnapshot();
  });
});
