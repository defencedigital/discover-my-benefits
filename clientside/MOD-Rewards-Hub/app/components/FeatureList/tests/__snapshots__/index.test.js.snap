// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureList component matches the snapshot 1`] = `ShallowWrapper {}`;
